webapp/src/serial.tsx

```diff
@@ -0,0 +1,251 @@
+import * as React from "react"
+import * as pkg from "./package"
+import * as core from "./core"
+import * as srceditor from "./srceditor"
+import * as sui from "./sui"
+import * as codecard from "./codecard"
+import * as canvaschart from "./canvaschart"
+
+import Cloud = pxt.Cloud
+import Util = pxt.Util
+
+const lf = Util.lf
+
+export class Editor extends srceditor.Editor {
+    //?
+    static counter = 0
+    private graphEntries: ILogEntryElement[] = []
+    private consoleEntries: IConsoleEntry[] = []
+    private consoleBuffer: string = ""
+
+    public element: HTMLDivElement
+    private labelElement: HTMLElement
+    private graphElement: HTMLElement
+    private consoleElement: HTMLElement
+
+    // TODO this!
+    // TODO pass these values in with props or config?
+    private shouldScroll = false
+    private isSim: boolean = true
+    private maxEntries: number = 80
+    private maxLineLength: number = 500
+    private maxAccValues: number = 500
+    private renderFiberId: number
+    private maxConsoleEntries: number = 20
+
+    private active: boolean = true
+
+    acceptsFile(file: pkg.File) {
+        // TODO hardcoded string
+        return file.name === "serial.json"
```

Comment:
define the constant here: https://github.com/Microsoft/pxt/blob/master/pxtlib/main.ts#L211

Reply:
Done